packages/react-dom-bindings/src/client/ReactDOMComponent.js

```diff
@@ -1364,28 +1449,59 @@ export function updateProperties(
           didWarnControlledToUncontrolled = true;
         }
       }
+
+      // Update checked *before* name.
```

Comment:
I believe this isn't the same as the behavior before. If you update from

```jsx
<input type="checkbox" name="a" checked={true} />
```
to

```jsx
<input type="radio" name="b" checked={false} />
```

then I believe now it will uncheck other `a` radios as well as other `b` radios (!) but the old code didn't. I think if only name and checked change then you didn't regress, but also I'm unsure how much it matters given the other bug I noticed where this handling doesn't really reliably work for name and checked anyway…?

The best strategy I could come up with that seemed obviously correct in all cases was to unset name before changing either type/checked and then set it back after. (You could also set checked to false but I was having trouble thinking about how that affects the dirty checkedness flag and interplay with the attribute.)

Reply:
Before it would always set checked before the loop. It would sometimes set it again inside the loop, before or after "name" depending on enumeration order of the props.

Now it will consistently always set checked first and name later.

Are you saying I should set `type` after the loop too?

I guess we just need to fix it properly.

Reply:
Basically it seems you want to avoid any intermediate states that have simultaneously `node.type = 'radio'` and `node.checked = true` and `node.name` nonempty. Which is tricky because you could be transitioning either from that or to that. I don't think any of the six possible (type, name, checked) permutations is always correct. So either you need to be really fancy about what values are changing and set them in different orders as a result, or you need to unset one of the properties, update the other two, then set the first.

Reply:
We already do that here:

https://github.com/facebook/react/blob/main/packages/react-dom-bindings/src/client/ReactDOMInput.js#L239

I think maybe this one is just here because of the weird case where `checked` is also set in the props - unlike value. It seems like maybe we should also use the special cases in `updateInput` and not update `checked` nor `type` in the loop.

Reply:
Oh, great. Yeah if we only set type and checked within that block then I think it's chill.